lrathoni/Lullaby-for-a-fish
# Added MOD music support.

### What

- **Currently supported only in Play mode, on Windows and Linux 64-bit platforms.**
- Added reading capabilities for MOD music through C# bindings of libopenmpt.
- Added Unity components capable of playing music through AudioSources.
- Added Unity component for dynamic playing.

### Known issues and to-dos

- Clicking occurs when stopping a Note with the playable instrument or music player.
- Support MacOS.
- Support Standalone for all plateforms.

### How to test

- Checkout this branch (see **Notes**).
- Open Project with Unity 2019.4
- Open the Scene under Packages/MptUnity/Assets/Scenes/DebugRoom_OpenMPT.
- Launch Play mode.
- Check that the instrument plays correctly (see **Notes**), without noticeable lag or distortion.
- Check that background music plays correctly (see **Notes**).

### Notes

- MptUnity was added as a submodule, some extra steps might be required in order to fetch it. Using GitKraken on Windows, this was done automatically, but I don't know about other clients. If Unity complains about missing scripts, open a command line interface (Powershell, git bash, linux shell or whatever), and write:

```
 git checkout add_mptunity
 git submodule update --init --recursive
```

- Currently, controls for the instruments are Y, U, I, O, P. They can be changed in the Flute Player script component of the Flute Player object.
- Background music can be played, pause, stop, with the blocks on the left of the scene.
- Beats Per Second (BPM) can be increased or lowered with the block in the centre.
- Musical Sections (parts of the whole music, *pupitres*, instrument ensembles), can be sected with the plus / minus blocks, and muted with the validation block.
